airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -353,6 +362,39 @@ public List<Job> listJobs(final Set<ConfigType> configTypes, final String config
         jobSelectAndJoin(jobsSubquery) + ORDER_BY_JOB_TIME_ATTEMPT_TIME)));
   }
 
+  @Override
+  public List<Job> listJobsIncludingId(final Set<ConfigType> configTypes, final String connectionId, final long targetJobId, final int pagesize)
```

Comment:
We don't have use case for `listJobsWithStatusIncludingJobId`? I am asking because it feels like we may just want to have a configurable query function rather than specific queries each time.
Some parts such as pagination should be common for example.
If this make sense, we can track it as a task later, doesn't have to be in this PR.

Reply:
I don't believe we have a use case for that today -- currently the only thing using `listJobsWithStatus` is the JobCleaner which I don't think is even running, and it doesn't have the same need of directly linking to a specific job that the frontend has